Mathlib/Tactic/Choose.lean

```diff
@@ -0,0 +1,240 @@
+/-
+Copyright (c) 2018 Johannes Hölzl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Floris van Doorn, Mario Carneiro, Reid Barton, Johan Commelin
+-/
+
+import Mathlib.Util.Asserts
+import Mathlib.Util.Tactic
+import Mathlib.Tactic.ShowTerm
+
+import Mathlib.Lean.getBinderName
+import Mathlib.Lean.mkFreshNameFrom
+import Mathlib.Lean.addLocalVarInfoForBinderIdent
+
+open Lean Lean.Meta Elab.Term Elab.Tactic
+
+/-!
+# `choose` tactic
+Performs Skolemization, that is, given `h : ∀ a:α, ∃ b:β, p a b |- G` produces
+`f : α → β, hf: ∀ a, p a (f a) |- G`.
+
+TODO: switch to `rcases` syntax: `choose ⟨i, j, h₁ -⟩ := expr`.
+-/
+namespace function
```

Comment:
This section presumably should live outside of `Tactic/`.

Reply:
It is really short, so I would actually like to keep it here. It makes the tactic self-contained, and independent of a defn in the main library. I've moved it inside that `Tactic.Choose` namespace to avoid conflicts.

Reply:
Okay. Could you just leave a comment to that effect (i.e. that it is sufficiently specific to the implementation of the tactic that it doesn't need to be moved)?

Reply:
Done